action.php

```diff
@@ -187,30 +189,137 @@ public function handle_parser_metadata_render(Doku_Event $event, $param)
         }
     }
 
+    /**
+     * Returns JSON with filtered users and groups
+     *
+     * @param Doku_Event $event
+     * @param string $param
+     */
+    public function handle_ajax_get(Doku_Event $event, $param)
+    {
+        if ($event->data != 'plugin_watchcycle_get') return;
+        $event->preventDefault();
+        $event->stopPropagation();
+        global $conf;
+
+        header('Content-Type: application/json');
+        try {
+            $result = $this->fetchUsersAndGroups();
+        } catch(\Exception $e) {
+            $result = [
+                'error' => $e->getMessage().' '.basename($e->getFile()).':'.$e->getLine()
+            ];
+            if($conf['allowdebug']) {
+                $result['stacktrace'] = $e->getTraceAsString();
+            }
+            http_status(500);
+        }
+
+        $json = new JSON;
```

Comment:
The JSON object is (or will be) deprecated. So I guess this should be changed into json_encode()

Reply:
You are right.